File: eCommerce/ClientApp/src/app/shared/classes/brands.ts
import { Operations } from "./enums/operations";

export class Brand {
 constructor(operation: Operations){
  this.Operation = operation
}

 id?: number;
 title?: string;
 shortName?: string;
 icon?: string;
 totalProductsCount?: number;
 isSelected?:boolean;

 series?: Series[];
 Operation?: Operations;
}

export class Series {
    constructor(operation: Operations) {
    this.Operation = operation
  }
 
  id?: number;
  title?: string;
  brandShortName?: string;
  shortName?: string;
  icon?: string;
  productsCount?: number;
  isSelected?:boolean;
 
  subBrands?: SubBrand[];
  Operation?: Operations;
 }
export class SubBrand {
    constructor(operation: Operations) {
    this.Operation = operation
  }
 
  id?: number;
  title?: string;
  brandShortName?: string;
  seriesShortName?: string;
  shortName?: string;
  icon?: string;
  productsCount?: number;
  isSelected?:boolean;
 
  Operation?: Operations;
 }
 

 
 

File: Entities/Models/Brands.cs
using Ardalis.GuardClauses;
using Entities.Enums;
using System.ComponentModel.DataAnnotations.Schema;

namespace Entities.Models
{
    public class Brands
    {
        public int Id { get; set; }
        public int TotalProductsCount { get; set; }

        public string Title { get; set; }
        public string Icon { get; set; }
        public string ShortName { get; set; }

        [NotMapped]
        public IEnumerable<Series> Series { get; set; }

        [NotMapped]
        public CRUD Operation { get; set; }

        [NotMapped]
        public ProductsQuantity ProductsQuantity { get; set; }

        public void Update(BrandInput Input)
        {
            Guard.Against.NullOrEmpty(Input.Title, nameof(Input.Title));
            Title = Input.Title;

            Guard.Against.NullOrEmpty(Input.ShortName, nameof(Input.ShortName));
            ShortName = Input.ShortName;

            Icon = Input.Icon;
            Icon = Input.Icon;
            TotalProductsCount = Input.TotalProductsCount;
        }
    }
        public class BrandProductRelation
    {
        public int Id { get; set; }

        public int ProductId { get; set; }
        public int BrandId { get; set; }

        public void Update(BrandProductRelationInput input)
        {
            Guard.Against.NegativeOrZero(input.ProductId, nameof(input.ProductId));
            ProductId = input.ProductId;

            Guard.Against.NegativeOrZero(input.BrandId, nameof(input.BrandId));
            BrandId = input.BrandId;
        }
    }
}
